src/components/ActivityForm.js

```diff
@@ -89,38 +100,51 @@ export default class ActivityForm extends React.Component {
   };
 
   render() {
-    const { name, day, classNo, room, teacher, error } = this.state;
+    const {
+      name,
+      day,
+      classNo,
+      room,
+      teacher,
+      classNoError,
+      nameError,
+      roomError,
+      teacherError,
+    } = this.state;
     return (
       <div>
-        {error && <p>{error}</p>}
         <form onSubmit={this.onSubmit}>
           <Input
             type="text"
             placeholder="Activity Name"
             value={name}
             onChange={this.onNameChange}
-            onBlur={this.onBlur}
+            onBlur={this.onBlur('nameError')}
+            errorMsg={nameError}
           />
           <Input
             type="text"
             placeholder="Teacher"
             value={teacher}
             onChange={this.onTeacherChange}
-            onBlur={this.onBlur}
+            onBlur={this.onBlur('teacherError')}
+            errorMsg={teacherError}
           />
           <Input
             type="text"
             placeholder="Room"
             value={room}
             onChange={this.onRoomChange}
-            onBlur={this.onBlur}
+            onBlur={this.onBlur('roomError')}
+            errorMsg={roomError}
           />
           <Input
             type="number"
             placeholder="Class Number"
             value={classNo}
             onChange={this.onClassNoChange}
-            onBlur={this.onBlur}
+            onBlur={this.onBlur('classNoError')}
```

Comment:
this is a very good way to make it properly. Why not handle in the same way the onChange?